### PR TITLE
Add delete reviews

### PR DESCRIPTION
**Pull request by JCWoodworker**
1. added ability to delete reviews that a user has submitted.
  a.  user can only delete THEIR reviews.
 2. Cleaned up some code in other areas.

### CODE REVIEW

File: server/src/routes/api/v1/servicesReviewsRouter.js [hunk 1]
```diff
@@ -23,4 +23,16 @@ servicesReviewsRouter.post("/", async (req,res) =>{
   }
 })
 
+servicesReviewsRouter.delete('/', async (req, res) => {
+  try {
+    const { reviewId } = req.body
+    const reviewToDelete = await Review.query().findById(reviewId)
+    await reviewToDelete.$relatedQuery('votes').delete()
```
Comment (JCWoodworker) [inline server/src/routes/api/v1/servicesReviewsRouter.js:30]: We had to delete the votes associated with the review in order to delete the review.  Votes on the "votes" table require a review to exist, so we received an error if we attempted to delete a review with more than 0 total votes.  Is there a better way to delete a row on one table and it's dependent rows from other tables?  Will the same issue happen if we add a feature to delete services?  (ie.  deleting service requires deleting dependent reviews then deleting dependent votes).

File: client/src/components/layout/ReviewTile.js [hunk 2]
```diff
@@ -2,7 +2,24 @@ import React from 'react'
...
+    if (props.user) {
+      if (props.user.id === props.review.userId) {
+        return(
+          <button onClick={handleDeleteClick}> 
+            Delete Review
+          </button>
+        )
+      }
+    }
+  }
```
Comment (nwalberts) [inline client/src/components/layout/ReviewTile.js:17]: `deleteReview` vs `DeleteYourReview` is starting to get confusing. I would rename this to be `deleteButtonElement`

Further, this doesnt need to be in a function at all. 

Instead, create a variable `deleteButtonElement` that is set to an empty string to begin. then

```js
if (props.user) {
      if (props.user.id === props.review.userId) {
         deleteButtonElement = <button onClick={handleDeleteClick}> 
            Delete Review
          </button>
      }
    }
```

Finally, add `deleteButtonElement` to your `ReviewTile` return

File: client/src/components/layout/ShowService.js [hunk 3]
```diff
@@ -35,12 +35,40 @@ const ShowService = (props) => {
...
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const updatedReviews = service.reviews.filter(review => review.id != reviewId)
+      setService({...service, reviews: updatedReviews})
+    } catch (error) {
+      return console.error(`Error in fetch: ${error.message}`)
+    }
+  }
```
Comment (nwalberts) [inline client/src/components/layout/ShowService.js:61]: very nice work

File: server/src/routes/api/v1/servicesReviewsRouter.js (hunk 1 above)
Comment (nwalberts) [reply server/src/routes/api/v1/servicesReviewsRouter.js:30]: This is good! We want this level of annoyance in our application. 

The only thing you may want to know in the future is that for a manyToMany relation, you may need to also use `unrelate`

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**